### PR TITLE
Clean Up Simulated Backend

### PR DESCRIPTION
**Pull request by nisdas**
To make it easier to reuse the simulated backend for testing on different services in Prysm, this PR will be cleaning up `backend` package so as to allow for more reusable components in running other tests, ex: Sync. 

- [x] Break up the large test functions into more modular, and easier to understand functions
~~- [ ] Add in any required tests~~

### CODE REVIEW

File: beacon-chain/chaintest/backend/simulated_backend.go [hunk 1]
```diff
@@ -262,32 +201,142 @@ func (sb *SimulatedBackend) RunStateTransitionTest(testCase *StateTestCase) erro
...
+		layersPeeledForProposer[uint64(idx)] = 0
+	}
+
+	return hashOnions, lastRandaoLayer, prevBlockRoots, layersPeeledForProposer, nil
+}
+
+// setupBeaconStateAndGenesisBlock creates the initial beacon state and genesis block in order to
+// proceed with the test.
+func (sb *SimulatedBackend) setupBeaconStateAndGenesisBlock(initialDeposits []*pb.Deposit) ([][32]byte, error) {
+
```
Comment (rauljordan) [inline beacon-chain/chaintest/backend/simulated_backend.go]: Eliminate this blank line at start of func

File: beacon-chain/chaintest/backend/simulated_backend.go [hunk 2]
```diff
@@ -262,32 +201,142 @@ func (sb *SimulatedBackend) RunStateTransitionTest(testCase *StateTestCase) erro
...
+	return &SimulatedObjects{
+		simDeposit:          simulatedDeposit,
+		simProposerSlashing: simulatedProposerSlashing,
+		simAttesterSlashing: simulatedAttesterSlashing,
+		simValidatorExit:    simulatedValidatorExit,
+	}
+}
+
+func (sb *SimulatedBackend) compareTestCase(testCase *StateTestCase) error {
```
Comment (rauljordan) [inline beacon-chain/chaintest/backend/simulated_backend.go]: Comment for this?

File: beacon-chain/chaintest/backend/simulated_backend.go [hunk 3]
```diff
@@ -262,32 +201,142 @@ func (sb *SimulatedBackend) RunStateTransitionTest(testCase *StateTestCase) erro
...
+	if err := sb.compareTestCase(testCase); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// initializeStateTest sets up the environment by generating all the required objects in order
+// to proceed with the state test.
+func (sb *SimulatedBackend) initializeStateTest(testCase *StateTestCase) (hashOnions [][32]byte,
```
Comment (rauljordan) [inline beacon-chain/chaintest/backend/simulated_backend.go]: Do these need to be methods of the simulated backend? It seems to me they look more like pure functions

Comment (prestonvanloon) [reply beacon-chain/chaintest/backend/simulated_backend.go]: It uses methods on the simulatedBackend, I think this appropriate

**Comment by nisdas** [comment]
@rauljordan PTAL , made the changes

**Pull request merged**